numbat/modules/math/functions.nbt

```diff
@@ -37,6 +37,11 @@ fn atanh(x: Scalar) -> Scalar
 
 fn gamma(x: Scalar) -> Scalar
 
+fn _qe_solution<A, B>(a: A, b: B, c: B²/A, sign: Scalar) -> B/A =
+  (-b + sign × sqrt(b² - 4 a c)) / 2 a
+fn solve_quadratic_equation<A2, B2>(a: A2, b: B2, c: B2²/A2) -> String =
+  "x₁ = {_qe_solution(a, b, c, 1)}; x₂ = {_qe_solution(a, b, c, -1)}"
```

Comment:
```suggestion
fn _qe_solution<A, B>(a: A, b: B, c: B²/A, sign: Scalar) -> B/A =
  (-b + sign × sqrt(b² - 4 a c)) / 2 a

fn solve_quadratic_equation<A, B>(a: A, b: B, c: B²/A) -> String =
  "x₁ = {_qe_solution(a, b, c, 1)}; x₂ = {_qe_solution(a, b, c, -1)}"
```

Reply:
Okay, replacing `A2` and `B2` with `A` and `B` respectively in `solve_quadratic_equation` causes this error:

```
error: while type checking                                                                                                                         
   ┌─ Module 'math::functions', File <builtin>/modules/math/functions.nbt:40:35                                                                    
   │                                
40 │ fn _qe_solution<A, B>(a: A, b: B, c: B²/A, sign: Scalar) -> B/A =   
   │                                   ^ Multiple unresolved generic parameters in a single function parameter type are not (yet) supported. Consid
er reordering the function parameters                                                                                                              
   ·                                                                                                                                               
47 │   else if b² <= 4 a c then if b² < 4 a c then "no real-valued solution" else "x = {-b / 2 a}" else "x₁ = {_qe_solution(a, b, c, 1)}; x₂ = {_qe
_solution(a, b, c, -1)}"                                                 
   │                                                                                                           ------------ In this function call  
                                                                         
Interpreter error in Prelude code 
```

@sharkdp I don't think this is supposed to happen?

Reply:
Ah this is https://github.com/sharkdp/numbat/issues/166